libs/blocks/table/table.js

```diff
@@ -483,64 +501,66 @@ function applyStylesBasedOnScreenSize(table, originTable) {
 
+    table.querySelectorAll('.hide-mobile').forEach((col) => { col.classList.remove('hide-mobile'); });
+
     if (isMerch && headingsLength > 2) {
-      table.querySelectorAll('.col:not(.col-1, .col-2)').forEach((col) => col.remove());
+      table.querySelectorAll('.col:not(.col-1, .col-2)').forEach((col) => { col.classList.add('hide-mobile'); });
     } else if (headingsLength > 3) {
-      table.querySelectorAll('.col:not(.col-1, .col-2, .col-3), .col.no-borders').forEach((col) => col.remove());
+      table.querySelectorAll('.col:not(.col-1, .col-2, .col-3), .col.no-borders').forEach((col) => { col.classList.add('hide-mobile'); });
     }
 
     if ((!isMerch && !table.querySelector('.col-3'))
       || (isMerch && !table.querySelector('.col-2'))) return;
 
-    const filterChangeEvent = () => {
-      table.innerHTML = originTable.innerHTML;
-      reAssignEvents(table);
+    const filterChangeEvent = (e) => {
       const filters = Array.from(table.parentElement.querySelectorAll('.filter')).map((f) => parseInt(f.value, 10));
       const rows = table.querySelectorAll('.row');
 
+      table.querySelectorAll('.hide-mobile, .force-last').forEach((col) => { col.classList.remove('hide-mobile', 'force-last'); });
+
+      rows.forEach((row) => {
+        const clonedCols = row.querySelectorAll('.col[data-cloned]');
+        clonedCols.forEach((col) => col.remove());
+      });
+
       if (isMerch) {
-        table.querySelectorAll(`.col:not(.col-${filters[0] + 1}, .col-${filters[1] + 1})`).forEach((col) => col.remove());
+        table.querySelectorAll(`.col:not(.col-${filters[0] + 1}, .col-${filters[1] + 1})`).forEach((col) => { col.classList.add('hide-mobile'); });
       } else {
-        table.querySelectorAll(`.col:not(.col-1, .col-${filters[0] + 1}, .col-${filters[1] + 1}), .col.no-borders`).forEach((col) => col.remove());
+        table.querySelectorAll(`.col:not(.col-1, .col-${filters[0] + 1}, .col-${filters[1] + 1}), .col.no-borders`).forEach((col) => { col.classList.add('hide-mobile'); });
       }
 
-      if (filters[0] > filters[1]) {
-        if (isMerch) {
-          rows.forEach((row) => row.querySelector('.col:not(.section-row-title)')
-            .classList.add('force-last'));
-        } else {
-          rows.forEach((row) => row.querySelector('.col:not(.section-row-title, .col-1)')
-            .classList.add('force-last'));
-        }
-      } else if (filters[0] === filters[1]) {
+      rows.forEach((row) => {
+        const secondFilterCol = row.querySelector(`.col-${filters[1] + 1}`);
+        if (secondFilterCol) secondFilterCol.classList.add('force-last');
+      });
+
+      if (filters[0] === filters[1]) {
+        const selectedCol = filters[0] + 1;
         rows.forEach((row) => {
-          row.append(row.querySelector('.col:last-child').cloneNode(true));
+          const selectedColumn = row.querySelector(`.col-${selectedCol}`);
+          if (!selectedColumn) {
+            return;
+          }
```

Comment:
We usually do one line returns 
```suggestion
          if (!selectedColumn) return;
```